File: src/defaultConfig.jl
include("keysym.jl")
include("core/tile.jl")
include("core/translationLayer.jl")

#TODO: config files must print error messages
#you can find names in keysym.jl
const mods = ["Shift_L", "Caps_Lock", "Control_L", "Alt_L", "", "", "Super_L", "ISO_Level3_Shift"]
const mod1 = mods[4]
#TODO what position in mods array?
const mod2 = "Num_Lock"
const mod3 = mods[2]
const mod4 = mods[7]
#also known as Alt Gr
const mod5 = mods[8]
const btnLeft = "Pointer_Button1"
const btnRight = "Pointer_Button2"
const btnMiddle = "Pointer_Button3"
const shift = mods[1]
const ctrl = mods[3]
#
@enum Screentransform begin
    NORMAL
end

function setLayout(l)
    global layout = l
    updateLayout()
end

#default
mod = mod1

File: src/defaultConfig.lua
require "tile"

-- TODO: config files must print error messages
-- you can find names in keysym.jl
mods = {"Shift_L", "Caps_Lock", "Control_L", "Alt_L", "", "", "Super_L", "ISO_Level3_Shift"}
mod1 = mods[4]
-- TODO what position in mods array?
mod2 = "Num_Lock"
mod3 = mods[2]
mod4 = mods[7]
-- also known as Alt Gr
mod5 = mods[8]
btnLeft = "Pointer_Button1"
btnRight = "Pointer_Button2"
btnMiddle = "Pointer_Button3"
shift = mods[1]
ctrl = mods[3]

cursorMode = {
    CurNormal = 0,
    CurMove = 1,
    CurResize = 2,
}

layouts = {
    { "[M]", function(n) monocle(n) end },
    { "[]=", function(n) tile(n) end },
    { "><>", function(n) floating(n) end },
}
layout = layouts[layoutId]

function setLayout()
    layoutId = layoutId + 1
    if layoutId > #layouts then
        layoutId = 1
    end
    mylib.updateLayout()
end

function setLayout(i)
    layoutId = i
    layout = layouts[i]
    action.updateLayout()
end

function toggleOverlay()
    print("toggle\n")
    -- setOverlay(!getOverlay())
    -- arrangeThis(false);
end

-- default
mod = mod1
